docs/app/pages/components/components-sections/select/select-custom/select-custom.component.html

```diff
@@ -0,0 +1,156 @@
+<div class="row">
+    <div class="col-md-5">
+
+        <ux-select-custom
+            [(selected)]="selected"
+            (filterChange)="filter = $event"
+            [allowNull]="allowNull"
+            [busy]="showBusyIndicator"
+            [width]="componentWidth"
+            [maxHeight]="componentMaxHeight"
+            [placeholder]="placeholder">
+
+            <ng-template #buttonContent>
+                <b>Selection:</b> {{ selected ? selected.name : '(none)' }}
+            </ng-template>
+
+            <ng-template #dropdownContent>
+
+                <div role="radiogroup"
+                     class="radio-button-container"
+                     tabindex="0"
+                     (keydown.space)="selectButton($event)"
+                     (keydown.up)="navigateUp($event)"
+                     (keydown.arrowUp)="navigateUp($event)"
+                     (keydown.down)="navigateDown($event)"
+                     (keydown.arrowDown)="navigateDown($event)">
+
+                    <ux-radio-button
+                        *ngFor="let option of optionList; let index = index"
+                        name="group"
+                        [value]="selected"
+                        (valueChange)="selected = $event"
+                        [option]="option"
+                        [hidden]="isHidden(option.name)"
+                        [class.focused]="focusIndex === index">
+                        {{ beforeHighlight(option.name) }}
```

Comment:
Instead of calling these three functions could we consider using a pipe as they are `pure` and avoid unnecessary change detection, as calling three functions for each item every time change detection runs could incur quite a performance hit on larger lists or heavy pages. We do something similar to this here: https://github.com/UXAspects/UXAspects/blob/develop-1.8.2/src/components/filters/filter-dynamic/filter-typeahead-highlight.pipe.ts

Reply:
I have the same (not-so-strong) opinion as before: This is demo code. I would prefer making it easy to understand over showing the consumer of the component how to implement optimal performance. If you still think it needs to be changed, I will do it.

Reply:
I do agree that it is a bit simpler, but I have found that very often people just copy the code from our examples verbatim, so if we do something like this they end up with slow pages once they have a page with 100s of items etc... So we tend to document with the best practices, that way people who blindly copy the code at least will have fast code

Reply:
Agree with this, sample code should always adhere to best practices.

Reply:
Thank you for your feedback! I have implemented the pipe now.